test/classes.coffee

```diff
@@ -1367,19 +1367,330 @@ test "nested classes with super", ->
 
     class @ExtendedInner extends @Inner
       constructor: ->
-        super()
-        # TODO this is invalid since `this` is accessed before `super`
-        # @label = super().label + ' extended'
+        tmp = super()
+        @label = tmp.label + ' extended'
 
-    extender: () ->
-      class @ExtendedSelf extends @
+    @extender: () =>
+       class ExtendedSelf extends @
         constructor: ->
-          super()
-          # TODO this is invalid since `this` is accessed before `super`
-          # @label = super + ' from this'
+          tmp = super()
+          @label = tmp.label + ' from this'
       new ExtendedSelf
 
   eq (new Outer).label, 'outer'
   eq (new Outer.Inner).label, 'inner'
-  # eq (new Outer.ExtendedInner).label, 'inner extended'
-  # eq (new Outer.extender()).label, 'inner from this'
+  eq (new Outer.ExtendedInner).label, 'inner extended'
+  eq (Outer.extender()).label, 'outer from this'
+
+# TODO: Currently this test fails.
+test "Static methods generate 'static' keywords", ->
+  compile = """
+  class CheckStatic
+    constructor: (@drink) ->
+    @className: 'CheckStatic'
```

Comment:
Only methods are emitted as `static`, non-methods are added to the executable body.